js/ui/handler/touch_zoom_rotate.js

```diff
@@ -55,9 +62,30 @@ TouchZoomRotate.prototype = {
             bearing = this._rotationDisabled ? 0 : vec.angleWith(this._startVec) * 180 / Math.PI,
             map = this._map;
 
+        this._scalingSignificantly =
+            this._scalingSignificantly || (Math.abs(1 - scale) > this._significantScaleThreshold);
+
+        this._rotatingSignificantly =
+            this._rotatingSignificantly || (Math.abs(bearing) > this._significantRotateThreshold);
+
+        // Similar to google maps: if the user's intent is pinch zooming, rotate
+        // will be disabled. If their intent is rotating, then both rotate and
+        // zoom will be supported. We determine 'intent' by whichever threshold is
+        // surpassed first. Once determined, we keep that state for the duration
+        // of this gesture.
```

Comment:
That is solved by setting it to undefined in the gesture start handler already. I opt'ed for making sure its reset at the start rather than the end. But we could do both to be more consistent?